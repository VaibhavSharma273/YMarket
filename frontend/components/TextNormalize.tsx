import { Dimensions, Platform, PixelRatio } from 'react-native';

const {
    width: SCREEN_WIDTH,
    height: SCREEN_HEIGHT,
  } = Dimensions.get('window');
  
  // based on iphone 5s's scale
  const scale = SCREEN_WIDTH / 350;

  export function normalize(size: number) {
    const newSize = size * scale 
    if (Platform.OS === 'ios') {
      return Math.round(PixelRatio.roundToNearestPixel(newSize))
    } else {
      return Math.round(PixelRatio.roundToNearestPixel(newSize)) - 2
    }
  }

export const text_styles = {
    mini: {
      fontSize: normalize(12),
    },
    small: {
      fontSize: normalize(15),
    },
    medium: {
      fontSize: normalize(17),
    },
    large: {
      fontSize: normalize(20),
    },
    xlarge: {
      fontSize: normalize(24),
    },
  };
