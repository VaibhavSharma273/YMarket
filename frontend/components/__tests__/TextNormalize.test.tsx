import * as React from 'react';
import renderer from 'react-test-renderer';
import { Text } from 'react-native';

import { normalize } from '../TextNormalize';

it(`renders correctly`, () => {
    const props = {
        fontSize: normalize(24),
    };
    const tree = renderer.create(<Text {...props}>Hi</Text>).toJSON();

    expect(tree).toMatchSnapshot();
});
