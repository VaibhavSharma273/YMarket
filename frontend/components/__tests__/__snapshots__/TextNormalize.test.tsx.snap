// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Text
  fontSize={52}
>
  Hi
</Text>
`;
